benchmark/criteo_1TB/preprocessing/train_eval_split.py

```diff
@@ -0,0 +1,142 @@
+    # file_handler.setLevel(logging.INFO)
+    # file_handler.setFormatter(logging.Formatter("[%(asctime)s]  [%(filename)15s:%(lineno)4d] %(levelname)-8s %(message)s"))
+    # logger.addHandler(file_handler)
+
+    for bin_file in bin_files:
+        bin_file_name = bin_file.name
+        original_file_wrapper = BinaryFileWrapper(
+            file_path=str(bin_file),
+            byteorder="little",
+            record_size=160,
+            label_size=4,
+        )
+        num_samples = original_file_wrapper.get_number_of_samples()
+        train_file_path = target_train_day_dataset_path / bin_file_name
+        test_file_path = target_test_day_dataset_path / bin_file_name
+
+        all_indices = list(range(num_samples))
+        random.seed(seed)
+        random.shuffle(all_indices)
+        test_size = max(int(num_samples * percentage / 100), 1)
+        test_indices = all_indices[:test_size]
+        train_indices = all_indices[test_size:]
+
+        logger.info(f"[worker {worker_id} at day {day}]: Splitting {bin_file_name} with {num_samples} into {len(train_indices)} training samples and {len(test_indices)} test samples")
+        original_file_wrapper.persist_sub_file(train_indices, str(train_file_path))
+        original_file_wrapper.persist_sub_file(test_indices, str(test_file_path))
+
+
+def main(
+    original_criteo_path: Annotated[pathlib.Path, typer.Argument(help="The path to the original criteo dataset")],
+    target_criteo_path: Annotated[pathlib.Path, typer.Argument(help="The path to save the split criteo dataset")],
+    seed: Annotated[int, typer.Argument(help="The seed to use for the random number generator")],
+    percentage: Annotated[int, typer.Option(help="The percentage of the dataset to use for evaluation")] = 1,
+    days_up_to: Annotated[Optional[int], typer.Option(help="Only split the dataset up to this day")] = None,
+    num_workers: Annotated[int, typer.Option(help="The number of workers to use for the split")] = 32,
+):
+    logger = logging.getLogger(__name__)
+    logger.info(f"Original Criteo path: {original_criteo_path}")
+    logger.info(f"Target Criteo path: {target_criteo_path}")
+    logger.info(f"Seed: {seed}")
+    logger.info(f"Percentage: {percentage}")
+    logger.info(f"Number of workers: {num_workers}")
+
+    # if the target path does not exist, create it
+    if not target_criteo_path.exists():
+        logger.info(f"Creating target Criteo path: {target_criteo_path}")
+        target_criteo_path.mkdir(parents=True)
+    # create the train and test directories
+    target_train_path = target_criteo_path / "train"
+    target_test_path = target_criteo_path / "test"
+    if not target_train_path.exists():
+        logger.info(f"Creating target train path: {target_train_path}")
+        target_train_path.mkdir(parents=True)
+    if not target_test_path.exists():
+        logger.info(f"Creating target test path: {target_test_path}")
+        target_test_path.mkdir(parents=True)
```

Comment:
i guess could also just do path.mkdir(exists_ok=True) but then we cannot print, whatever you prefer